array/DNAm/analysis/EWAS/lmWithinCT.r

```diff
@@ -0,0 +1,121 @@
+	modelFull<-lm(row ~ QCmetrics$Phenotype + QCmetrics$Cell.Proportion + QCmetrics$CETYGO + QCmetrics$CCDNAmAge + QCmetrics$Sex + QCmetrics$Tissue.Centre)
+	modelCC <- lm(row ~ QCmetrics$Phenotype + QCmetrics$Cell.Proportion + QCmetrics$CCDNAmAge + QCmetrics$Sex + QCmetrics$Tissue.Centre)
+  # extract case control & cell composition effects
+  return(c(
+    summary(modelFull)$coefficients["QCmetrics$PhenotypeSchizophrenia",c(1,2,4)],
+    summary(modelFull)$coefficients["QCmetrics$Cell.Proportion",c(1,2,4)],
+	summary(modelFull)$coefficients["QCmetrics$CETYGO",c(1,2,4)],
+	summary(modelCC)$coefficients["QCmetrics$PhenotypeSchizophrenia",c(1,2,4)],
+    summary(modelCC)$coefficients["QCmetrics$Cell.Proportion",c(1,2,4)],
+	summary(modelNull)$coefficients["QCmetrics$PhenotypeSchizophrenia",c(1,2,4)]))	
+}
+
+runEWASbasic<-function(row,QCmetrics){
+   modelNull<-lm(row ~ QCmetrics$Phenotype +  QCmetrics$CCDNAmAge + QCmetrics$Sex + QCmetrics$Tissue.Centre) 
+   modelFull<-lm(row ~ QCmetrics$Phenotype + QCmetrics$CETYGO + QCmetrics$CETYGO + QCmetrics$CCDNAmAge + QCmetrics$Sex + QCmetrics$Tissue.Centre)
+	
+	  # extract case control & cell composition effects
+  return(c(
+    summary(modelFull)$coefficients["QCmetrics$PhenotypeSchizophrenia",c(1,2,4)],
+	summary(modelFull)$coefficients["QCmetrics$CETYGO",c(1,2,4)],
+	summary(modelNull)$coefficients["QCmetrics$PhenotypeSchizophrenia",c(1,2,4)]))	
+
+}
+
+
+
+#----------------------------------------------------------------------#
+# LOAD PACKAGES
+#----------------------------------------------------------------------#
+
+library(doParallel)
+
+#----------------------------------------------------------------------#
+# DEFINE PARAMETERS
+#----------------------------------------------------------------------#
+
+args<-commandArgs(trailingOnly = TRUE)
+dataDir <- args[1]
+cellType <- args[2]
+
+normData<-file.path(dataDir, "3_normalised/normalised.rdata")
+cellCompData<-file.path(dataDir, "4_analysis", "EstimatedNeuralCellComposition.rdata")
+resPath<-file.path(dataDir, "4_analysis/EWAS")
+
+#----------------------------------------------------------------------#
+# LOAD AND PREPARE DATA
+#----------------------------------------------------------------------#
+
+setwd(dataDir)
+load(normData)
+load(cellCompData)
+
+print(paste0("running EWAS on ", cellType, " cell type..."))
+## subset to cell type samples
+QCmetrics<-QCmetrics[which(QCmetrics$Cell.type == cellType),]
+
+# filter to schizophrenia and control only
+QCmetrics<-QCmetrics[QCmetrics$Phenotype %in% c("Schizophrenia", "Control"),]
+
+
+# add cell composition proportions & CETYGO scores
+
+if (cellType == "Double-"){
+   QCmetrics$Cell.Proportion <- predPropBest[QCmetrics$Basename, "NeuNNeg_Sox10Neg_IRF8Pos"]
+} else if (cellType == "NeuN+"){
+  QCmetrics$Cell.Proportion <- predPropBest[QCmetrics$Basename, "NeuNPos_SOX6Pos"]
+} else if (cellType == "Sox10+"){
+  QCmetrics$Cell.Proportion <- NA

```

Comment:
Maybe this is obvious, but I can't see it. Why is the cell proportion being set to `NA` here?